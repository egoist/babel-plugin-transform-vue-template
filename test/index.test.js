const fs = require('fs')
const path = require('path')
const babel = require('@babel/core')

const plugin = require.resolve('../')

test('main', () => {
  const { code } = babel.transform(
    fs.readFileSync(path.join(__dirname, 'fixtures/a.js'), 'utf8'),
    {
      babelrc: false,
      plugins: [plugin]
    }
  )

  expect(code).toMatchSnapshot()
})
