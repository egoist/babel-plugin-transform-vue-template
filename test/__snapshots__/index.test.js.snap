// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
"const Component = {
  render: function render() {
    var _vm = this;

    var _h = _vm.$createElement;

    var _c = _vm._self._c || _h;

    return _c('div', [_vm._m(0), _vm._v(\\" \\"), _vm.message ? _c('p', [_vm._v(\\"\\\\n      \\" + _vm._s(_vm.message) + \\"\\\\n    \\")]) : _c('p', [_vm._v(\\"\\\\n      No message.\\\\n    \\")])]);
  },
  staticRenderFns: [function () {
    var _vm = this;

    var _h = _vm.$createElement;

    var _c = _vm._self._c || _h;

    return _c('header', [_c('h1', [_vm._v(\\"I'm a template!\\")])]);
  }]
};
const Another = {
  // @transform-disable
  template: \`<div></div>\`
};
const Yet = {
  render: function render() {
    var _vm = this;

    var _h = _vm.$createElement;

    var _c = _vm._self._c || _h;

    return _c(\\"div\\");
  },
  staticRenderFns: []
};
const Issue_11 = {
  render: function render() {
    var _vm = this;

    var _h = _vm.$createElement;

    var _c = _vm._self._c || _h;

    return _c('ABC');
  },
  staticRenderFns: []
};
const htmlTag = {
  render: function render() {
    var _vm = this;

    var _h = _vm.$createElement;

    var _c = _vm._self._c || _h;

    return _c('div', [_vm._v(\\"\\\\n    \\" + _vm._s(_vm.hehe) + \\"\\\\n  \\")]);
  },
  staticRenderFns: []
};"
`;
